### PR TITLE
GateKeeper: Remove verifyNotNull(…) #7840

### LINKED ISSUE #7840: GateKeeper: Remove verifyNotNull(…)
`verifyNotNull(Object, String)` is a precondition check which checks whether the `Object` parameter is null. If it is null, an `UnauthorizedAccessException` with the error message `"Trying to access system using a non-existent entity"` will be thrown. 

This doesn’t seem right as the check simply checks whether the variable is null or not; there is no connotation of unauthorised access.

Let’s change callers to use Guava’s `checkNotNull(Object, String)` (see #7730) instead, and let’s remove `verifyNotNull(…)`.

### PR DESCRIPTION
**Pull request by ParmaarK**
Fixes #7840

### CODE REVIEW

File: src/main/java/teammates/logic/api/GateKeeper.java [hunk 1]
```diff
@@ -293,7 +293,7 @@ public void verifyAccessible(InstructorAttributes instructor, FeedbackSessionAtt
      */
 
     // TODO: to be implemented when we adopt more finer-grain access control.
-    private void verifyNotNull(Object object, String typeName) {
+    private void checkNotNull(Object object, String typeName) {
```
Comment (whipermr5) [inline src/main/java/teammates/logic/api/GateKeeper.java]: Please read the issue again; it's not about renaming a method.

Comment (ParmaarK) [inline src/main/java/teammates/logic/api/GateKeeper.java]: Hi, can you guide me through this? 
I read the thread and found this : 

> This I agree. I'm wondering whether we can have a workaround. That is, whenever we encounter NPE, we log it down first before throwing NPE. That way, we can effectively find out which object caused the error (just have a counter to count which item in the varargs that it's checking). That way we need our own implementation of checkNotNull though hmm. (It's simple to implement though).

**Comment by damithc** [comment]
@ParmaarK the issue is not accepted yet. You may want to pick a different one until we decide whether to go ahead with the issue and how exactly to do it.

**Comment by ParmaarK** [comment]
@damithc sure. Thanks :)

**Comment by whipermr5** [comment]
Closing as the issue has been closed.

**Pull request closed without merging**